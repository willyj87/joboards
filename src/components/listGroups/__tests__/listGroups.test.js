import React from 'react';
import renderer from 'react-test-renderer';
import * as uuid from 'uuid';
import { render, screen, fireEvent } from '@testing-library/react';
import ListGroups from '../index';
import items from './__mock__/items.mock';

jest.mock('uuid');

const totalOffers = 4;
const handleClick = jest.fn();
const Component = <ListGroups items={items} totalOffers={totalOffers} handleClick={handleClick} />;
describe('ListGroups', () => {
  it('renders ListGroups correctly', () => {
    uuid.v4.mockReturnValue('1234');
    const tree = renderer
      .create(Component)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('display correct badges', () => {
    render(Component);
    const badge = screen.getByTestId('all-test-badge');
    expect(badge).toHaveTextContent('4');
  });
  it('should call the handleClick method', () => {
    render(Component);
    const listGroupsItem = screen.getByTestId('Business Intelligence & Machine Learning');
    fireEvent.click(listGroupsItem);
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
