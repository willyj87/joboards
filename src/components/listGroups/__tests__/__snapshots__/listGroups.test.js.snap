// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListGroups renders ListGroups correctly 1`] = `
<div
  className="list-group list-group-flush"
  onKeyDown={[Function]}
>
  <button
    className="d-flex justify-content-between align-items-start list-group-item active list-group-item-action"
    data-rr-ui-event-key="0"
    onClick={[Function]}
  >
    <div
      className="me-auto"
      data-testid="all-test"
    >
      all
    </div>
    <span
      className="badge rounded-pill bg-primary"
      data-testid="all-test-badge"
      variant="success"
    >
      4
    </span>
  </button>
  <button
    className="d-flex justify-content-between align-items-start list-group-item list-group-item-action"
    data-rr-ui-event-key="Business Intelligence & Machine Learning_1234"
    data-testid="Business Intelligence & Machine Learning"
    onClick={[Function]}
  >
    <div
      className="me-auto"
    >
      Business Intelligence & Machine Learning
    </div>
    <span
      className="badge rounded-pill bg-primary"
      data-testid="Business Intelligence & Machine Learning-badge"
      variant="success"
    />
  </button>
</div>
`;
